File: Darts/src/code/db/sanity/SanityCheckResultSimpleTableModel.java
package code.db.sanity;

import javax.swing.table.DefaultTableModel;

import util.TableUtil.DefaultModel;

public class SanityCheckResultSimpleTableModel extends AbstractSanityCheckResult
{
	private String description = null;
	private DefaultModel model = null;
	
	public SanityCheckResultSimpleTableModel(DefaultModel model, String description)
	{
		this.model = model;
		this.description = description;
	}

	@Override
	public String getDescription()
	{
		return description;
	}

	@Override
	public int getCount()
	{
		return model.getRowCount();
	}

	@Override
	protected DefaultTableModel getResultsModel()
	{
		return model;
	}
}

File: src/main/kotlin/dartzee/db/sanity/SanityCheckResultSimpleTableModel.kt
package dartzee.db.sanity

import dartzee.core.util.TableUtil.DefaultModel

open class SanityCheckResultSimpleTableModel(
    private val model: DefaultModel,
    private val desc: String
) : AbstractSanityCheckResult() {
    override fun getDescription() = desc

    override fun getCount() = model.rowCount

    override fun getResultsModel() = model
}
